components/prism-core.js

```diff
@@ -139,19 +139,24 @@ var _ = _self.Prism = {
 		},
 
 		// Traverse a language definition with Depth First Search
-		DFS: function(o, callback, type, visited) {
+		DFS: function DFS(o, callback, type, visited) {
 			visited = visited || {};
 			for (var i in o) {
 				if (o.hasOwnProperty(i)) {
 					callback.call(o, i, o[i], type || i);
 
-					if (_.util.type(o[i]) === 'Object' && !visited[_.util.objId(o[i])]) {
-						visited[_.util.objId(o[i])] = true;
-						_.languages.DFS(o[i], callback, null, visited);
+					var property = o[i],
+					    propertyType = _.util.type(property);
+
+					var objId = _.util.objId;
+
+					if (propertyType === 'Object' && !visited[objId(property)]) {
```

Comment:
Do we want to cache `objId(property)` instead of just the `objId` function?

Reply:
I just wanted to cache the function itself but now that I look at it, we might as well cache the result.

Reply:
I tried to cache the result and you can't really do it without producing ugly code. 
That's because you can only call `objId(property)` after you know that it's an object/array, otherwise, `Object.defineProperty` will throw an error.